internal/locate/store_cache.go

```diff
@@ -386,6 +387,36 @@ func (s resolveState) String() string {
 	}
 }
 
+// initByStoreMeta initializes the store fields by the given store meta. It should be protected by `resolveMutex`.
+func (s *Store) initByStoreMeta(store *metapb.Store) error {
+	if store == nil || store.GetState() == metapb.StoreState_Tombstone {
+		// The store is a tombstone.
+		s.setResolveState(tombstone)
+		return nil
```

Comment:
Should we clean up `s.addr` while the store is a tombstone?

I see it will return `s.addr, nil` when this function returns no error.